### PR TITLE
Add new column to tasks table, add value for poa change to enum

### PR DESCRIPTION
**Pull request by bgantick**
Resolves [CASEFLOW-1427](https://vajira.max.gov/browse/CASEFLOW-1427)

### Description
Adds a `cancellation_reason` column to the `tasks` table, for the purpose of tracking the reason for the most recent `cancelled` status. Based on [cross team discussion on 4/23/2021](https://hackmd.io/uLVtCCmSSIy6iUS54_FIgQ).

### Code Documentation Updates
- [x] [Wiki](https://github.com/department-of-veterans-affairs/caseflow/wiki/Tasks#cancellation-reason) updated

### Database Changes
*Only for Schema Changes*

* [x] Timestamps (created_at, updated_at) for new tables
* [x] Column comments updated
* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
* [x] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [x] ~Query profiling performed (eyeball Rails log, check bullet and fasterer output)~
* [x] Appropriate indexes added (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [x] DB schema docs updated with `make docs` (after running `make migrate`)
* [x] #appeals-schema notified with summary and link to this PR
* [x] ~Any non-obvious semantics or logic useful for interpreting database data is documented at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)~

### CODE REVIEW

File: db/migrate/20210426181017_add_cancellation_reason_to_tasks.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AddCancellationReasonToTasks < Caseflow::Migration
+  def change
+    add_column :tasks, :cancellation_reason, :string,
+               comment: "Reason for latest cancellation status"
+  end
+end
```
Comment (Jeremy1026) [inline db/migrate/20210426181017_add_cancellation_reason_to_tasks.rb:6]: Are there any indexes that would make sense to exist for this column?

**Review by Jeremy1026 — APPROVED** [review]

File: db/migrate/20210426181017_add_cancellation_reason_to_tasks.rb (hunk 1 above)
Comment (bgantick) [reply db/migrate/20210426181017_add_cancellation_reason_to_tasks.rb:7]: Added

**Comment by yoomlam** [comment]
Please add use cases for this new column to https://github.com/department-of-veterans-affairs/caseflow/wiki/Tasks#task-status

**Comment by bgantick** [comment]
@yoomlam New section added to wiki here: https://github.com/department-of-veterans-affairs/caseflow/wiki/Tasks#cancellation-reason

**Review by yoomlam — APPROVED** [review]
👍

**Pull request merged**